### DISCUSSION TITLE
Windows rearranged after monitors turned off

### DISCUSSION THREAD

**Discussion by Techngro** [kangyu-california/PersistentWindows#207] [Q&A]
Hi. Thanks for this program, I have been using it for a while and it's very useful. 

I have a strange issue where some of the windows become rearranged when I turn my monitors (Vizio 4K TV using HDMI, 1080P BenQ using DVI) back on. I checked in Event Log and it looks like my 4K monitor is being removed from the display settings? This is the general order of the messages I see multiple times:

```
PersistentWindows: Display settings changed Display_Loc0x0_Res1920x1080

PersistentWindows: Restore halted due to new display setting change Display_Loc0x0_Res1920x1080 

PersistentWindows: Restart restore for Display_Loc0x0_Res3802x2138__Display_Loc3802x0_Res1920x1080

PersistentWindows: Display settings changed Display_Loc0x0_Res3802x2138__Display_Loc3802x0_Res1920x1080

PersistentWindows: Start restoring window layout back to 2/15/2023 12:26:57 PM for display setting Display_Loc0x0_Res3802x2138__Display_Loc3802x0_Res1920x1080

PersistentWindows: Display settings changed Display_Loc0x0_Res1920x1080

PersistentWindows: no need to restore fresh session Display_Loc0x0_Res1920x1080

PersistentWindows: Display settings changed Display_Loc0x0_Res3802x2138__Display_Loc3802x0_Res1920x1080

PersistentWindows: Start restoring window layout back to 2/15/2023 12:26:57 PM for display setting Display_Loc0x0_Res3802x2138__Display_Loc3802x0_Res1920x1080

PersistentWindows: Restore finished in pass 2 with 19 windows recovered for display setting Display_Loc0x0_Res3802x2138__Display_Loc3802x0_Res1920x1080

```

I should also note that I am using DisplayFusion as a window manager. But that doesn't seem to be the issue because some of the windows never have a problem being restored (the same windows every time).

Any help would be appreciated.

**Comment by kangyu-california** [comment]
The issue is caused by restore window started too early while the TV/monitor is still undergoing power-recycle process.
The workaround is to run PW with -delay_auto_restore 5 to give the Tv/monitor enough time to stabilize.
You may adjust the delay value in seconds as needed.

**Reply by Techngro** [reply]
Ok. Thanks for the response. 

> I previously had the task setup with "-delay_start 5" in the 'Add arguments' section of Action. I will change to 
> "-delay_auto_restore 5" and see if that works.

Actually, I will keep the original argument and add the new one.

**Reply by Techngro** [reply]
Unfortunately, this did not fix the issue. I am still getting 

> PersistentWindows: Display settings changed Display_Loc0x0_Res1920x1080
> 
> PersistentWindows: Restore halted due to new display setting change Display_Loc0x0_Res1920x1080

in event viewer and the windows are still being rearranged. 

In my task scheduler, in the 'Action' setting, I have -delay_start 10, -delay_auto_restore 10 in 'Add arguments'. Is this the correct way to do it?

**Comment by kangyu-california** [comment]
That is right, it is better to remove the comma ',' between multiple arguments, just use space as separator

-delay_start 10 -delay_auto_restore 10

TV requires much longer time to wakeup from sleep than monitor.
